pallets/loans/src/lib.rs

```diff
@@ -681,16 +711,14 @@ impl<T: Config> TPoolNav<PoolIdOf<T>, T::Balance> for Pallet<T> {
 
 #[macro_export]
 macro_rules! ensure_role {
-	( $pool_id:expr, $origin:expr, $role:expr $(,)? ) => {{
-		let sender = ensure_signed($origin)?;
+	( $pool_id:expr, $sender:expr, $role:expr $(,)? ) => {{
```

Comment:
I have modified this macro because it tied two concepts (ensuring the origin and ensuring the role) that are not at first related. For example, the new changes check the origin once but check for more than one role.

Also, maybe this macro no longer needs to be a macro and can be a normal function that returns an error, similar to `ensure_signed()?` works.

Reply:
I like that!